salt/cert/init.sls

```diff
@@ -38,14 +38,14 @@ include:
     - signing_policy: minion
     - public_key: /etc/pki/minion.key
     - CN: {{ grains['fqdn'] }}
-    - C: {{ pillar['certificate_information']['subject_properties']['C'] }}
-    - Email: {{ pillar['certificate_information']['subject_properties']['Email'] }}
-    - GN: {{ pillar['certificate_information']['subject_properties']['GN'] }}
-    - L: {{ pillar['certificate_information']['subject_properties']['L'] }}
-    - O: {{ pillar['certificate_information']['subject_properties']['O'] }}
-    - OU: {{ pillar['certificate_information']['subject_properties']['OU'] }}
-    - SN: {{ pillar['certificate_information']['subject_properties']['SN'] }}
-    - ST: {{ pillar['certificate_information']['subject_properties']['ST'] }}
+    - C: "{{ pillar['certificate_information']['subject_properties']['C'] }}"
+    - Email: "{{ pillar['certificate_information']['subject_properties']['Email'] }}"
+    - GN: "{{ pillar['certificate_information']['subject_properties']['GN'] }}"
+    - L: "{{ pillar['certificate_information']['subject_properties']['L'] }}"
+    - O: "{{ pillar['certificate_information']['subject_properties']['O'] }}"
+    - OU: "{{ pillar['certificate_information']['subject_properties']['OU'] }}"
+    - SN: "{{ pillar['certificate_information']['subject_properties']['SN'] }}"
+    - ST: "{{ pillar['certificate_information']['subject_properties']['ST'] }}"
```

Comment:
`yaml_dquote` is not for embeding YAML code but for escaping some text and produce some valid YAML code. From the documentation: `yaml_dquote`: _"Serializes a string into a properly-escaped YAML double-quoted string. This is useful when the contents of a string are unknown and may contain quotes or unicode that needs to be preserved. The resulting string will be emitted with opening and closing double quotes."_

Not strictly necessary in this case (as you said, `"` should be enough), but it also escapes unicode and so...